src/gsi/read_prepbufr.f90

```diff
@@ -1132,6 +1135,11 @@ subroutine read_prepbufr(nread,ndata,nodata,infile,obstype,lunout,twindin,sis,&
               if (howvob)             call ufbint(lunin,owave,1,255,levs,owavestr)
               if (cldchob)            call ufbint(lunin,cldceilh,1,255,levs,cldceilhstr)
            endif
+!          Extract obs of howv (significant wave height, aka howv,
+!                               and howv must be listed in anavinfo and in firstguess)
+           if ( i_howv_in_anav == 1 .and. i_howv_in_data == 1 ) then
+              if (howvob)             call ufbint(lunin,owave,1,255,levs,owavestr)
+           endif
```

Comment:
Yes, I agree. 
Originally I did not add the line 
1140      if ( i_howv_in_anav == 1 .and. i_howv_in_data == 1 ) then

But later I noticed the previous if-block 1133-1137 for "twodvar_regional" (ie. 2DRTMA). When running for 2DRTMA, since twodvar_regional=.true., the line 1135 will decode owave for howbob, then after this if-block, line 1141 will decode the owave for howvob again. It might be OK to decode the owave twice in 2DRTMA run. But I am not sure about this, and defiantly I would not like to introduce potential problem into operational 2DRTMA run. 
So I add line 1140 -- "if ( i_howv_in_anav == 1 .and. i_howv_in_data == 1 ) then" to decode owave when running in 3D analysis (non-2DRTMA), eg 3DRTMA, or RRFS-DA.  2DRTMA does not check i_howv_in_data, so it will not run into this if-block and decode owave one more time.

Reply:
If we are on the same page, just remove line 1138-1142, move line 1135 out of the if block and then we are good.

Reply:
Yes, I agree. I thought about this before. But I do not know whether Manuel would like me to touch the 2DRTMA code. I can ask him when he is back from vacation.